sandbox/PLASS/explore/clustering_snakemake/clustering-workflow.snakefile

```diff
@@ -8,82 +6,71 @@ import pysam
-#FAA = "all_loosetrim.plass.c100"
-#OUT_BASE = "plass-loosetrim-PF00204-hmmscanT100"
-
-#PFAM_LINK = "https://pfam.xfam.org/family/PF00181/alignment/full" # rplb
-#PFAM_BASE = "PF00181_full_rplb"
-#FAA = "all_hardtrim.plass.c100"
-#OUT_BASE = "plass-hardtrim-PF00181-hmmscanT100"
-#FAA = "all_loosetrim.plass.c100"
-#OUT_BASE = "plass-loosetrim-PF00181-hmmscanT100"
-
-#PFAM_LINK = "https://pfam.xfam.org/family/PF00189/alignment/full" # rpsc
-#PFAM_BASE = "PF00189_full_rpsc"
-#FAA = "all_hardtrim.plass.c100"
-#OUT_BASE = "plass-hardtrim-PF00189-hmmscanT100"
-#FAA = "all_loosetrim.plass.c100"
-#OUT_BASE = "plass-loosetrim-PF00189-hmmscanT100"
-
-#PFAM_LINK = "https://pfam.xfam.org/family/PF00154/alignment/full" # recA
-#PFAM_BASE = "PF00154_full_reca"
-#FAA = "all_hardtrim.plass.c100"
-#OUT_BASE = "plass-hardtrim-PF00154-hmmscanT100"
-#FAA = "all_loosetrim.plass.c100"
-#OUT_BASE = "plass-loosetrim-PF00154-hmmscanT100"
-
-#PFAM_LINK = "https://pfam.xfam.org/family/PF01411/alignment/full" # alaS
-#PFAM_BASE = "PF01411_full_alas"
-#FAA = "all_hardtrim.plass.c100"
-#OUT_BASE = "plass-hardtrim-PF01411-hmmscanT100"
-#FAA = "all_loosetrim.plass.c100"
-#OUT_BASE = "plass-loosetrim-PF01411-hmmscanT100"
-
-#PFAM_LINK = "https://pfam.xfam.org/family/PF00562/alignment/full" # rpb2 d6
-#PFAM_BASE = "PF00562_full_rpb2d6"
-#FAA = "all_hardtrim.plass.c100"
-#OUT_BASE = "plass-hardtrim-PF00562-hmmscanT100"
-#FAA = "all_loosetrim.plass.c100"
-#OUT_BASE = "plass-loosetrim-PF00562-hmmscanT100"
+
+TRIM = ["hard"]
+# TRIM = ["loose"]
+
+#OUT_BASE = ["plass-hardtrim-all-bin-PF00521-hmmscanT100",
+#            "plass-hardtrim-all-bin-PF00204-hmmscanT100",
+#            "plass-hardtrim-all-bin-PF00181-hmmscanT100",
+#            "plass-hardtrim-all-bin-PF00189-hmmscanT100",
+#            "plass-hardtrim-all-bin-PF00154-hmmscanT100",
+#            "plass-hardtrim-all-bin-PF01411-hmmscanT100",
+#            "plass-hardtrim-all-bin-PF00562-hmmscanT100"]
+
+#OUT_BASE = ["plass-loosetrim-all-bin-PF00521-hmmscanT100",
+#            "plass-loosetrim-all-bin-PF00204-hmmscanT100",
+#            "plass-loosetrim-all-bin-PF00181-hmmscanT100",
+#            "plass-loosetrim-all-bin-PF00189-hmmscanT100",
+#            "plass-loosetrim-all-bin-PF00154-hmmscanT100",
+#            "plass-loosetrim-all-bin-PF01411-hmmscanT100",
+#            "plass-loosetrim-all-bin-PF00562-hmmscanT100"]
 
 rule all:
     input: 
-        expand("outputs/pid/{out_base}-mds.csv", out_base = OUT_BASE)
- 
+        expand("outputs/pid/plass-{trim}trim-all-bin-{pfam_base}-hmmscanT100-mds.csv", trim = TRIM, pfam_base = PFAM_BASE)
+
+
 rule download_pfam:
     output: "inputs/pfam/{pfam_base}.sto"
     params: 
-        pfam_link = PFAM_LINK,
-        pfam_base = PFAM_BASE,
+        #pfam_link = PFAM_LINK,
+        #pfam_base = PFAM_BASE,
         out_dir = "inputs/pfam"
     shell:'''
-    wget -O {params.out_dir}/{params.pfam_base}.sto {params.pfam_link}
+    pfam=$(echo {wildcards.pfam_base} | cut -f1 -d"_")
```

Comment:
you can also do this in the `params`, might be clearer in Python than using `echo|cut`

Reply:
more clear to everyone except me :) just kidding, I'll consider doing that! Thank you!!